<!DOCTYPE html>
<html lang="" xml:lang="">
  <head>
    <title>Trial Lecture</title>
    <meta charset="utf-8" />
    <meta name="author" content="John Zobolas   PhD Candidate" />
    <script src="libs/header-attrs-2.9/header-attrs.js"></script>
    <link href="libs/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="libs/remark-css-0.0.1/default-fonts.css" rel="stylesheet" />
    <link href="libs/font-awesome-5.13.0/css/all.min.css" rel="stylesheet" />
    <link href="libs/font-awesome-5.13.0/css/v4-shims.min.css" rel="stylesheet" />
    <link rel="stylesheet" href="css/ntnu.css" type="text/css" />
    <link rel="stylesheet" href="css/my.css" type="text/css" />
    <link rel="stylesheet" href="css/title.css" type="text/css" />
    <link rel="stylesheet" href="css/code.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">
class: center, middle, inverse, title-slide

# Trial Lecture
## Assets and drawbacks of commonly used computational frameworks in cancer modeling
### John Zobolas <br/> PhD Candidate
### Department of Biology, NTNU
### August 10th, 2021

---


layout: true

&lt;div class="my-footer"&gt;
&lt;span&gt;
  &lt;img src="img/ntnu/ntnu-logo-neg.png" style="width:160px;height:30px;"&gt;
&lt;/span&gt;
&lt;/div&gt;

---
class: middle

# Outline

## 1) .large.blue[Introduction to Cancer Modeling]
----
## 2) .large.blue[Computational Frameworks in practice]
----
## 3) .large.blue[Summary]

???

- Present **some** frameworks, not all of them! 
- Use practical examples/applications and then discuss the associated frameworks
- Point is to **state the differences** and **highlight drawbacks and assets of each discussed framework in practice**
- Target audience is Master-level students

---
class: center, middle

# .larger.blue[Introduction to Cancer Modeling]
----
&lt;img src="https://media.giphy.com/media/680qJiuIMtNle/source.gif"&gt;&lt;/img&gt;

---

### Why study cancer?

.red[2nd cause of death] worldwide 

.small.purple[Ritchie &amp; Roser (2018). Causes of Death. https://ourworldindata.org/causes-of-death]

### Why is cancer challenging?

![:scale 81%](img/trial-lecture/cancer_challenges.svg)

???

### Why study cancer?

- Cancer problem exacerbates due to aging and population growth
- Curing every cancer type is impossible

### Cancer is challenging

- The cancer hallmarks .purple[Hanahan &amp; Weinberg (2011)] as a guide to understand the complex processes that drive cancer
- Research findings are **not translational**
  - *In-vitro* and *in-vivo* approaches are .red[not effective] in the clinic
  - .red[&lt; 4% of drugs succeed] in clinical trials .small.purple[Wong (2019)]
  - Need for .green[basic science] and .green[integrative (in-silico) approaches]
- **High complexity**
  - Multiscale, dynamically adaptive process
  - Heterogeneity .small.purple[Senft (2017)]
      - Intra-tumor (within a patient)
      - Inter-tumor (patients diversity)
- **Acquired resistance to therapy**
- **Lack of biomarkers** that correlate with patient response
  - Need strategies for .green[patient stratification]
- **Lack or constrained access** to experimental and clinical data

---

# How to study cancer?

.pull-left-58[
### The Scientific Method

![](img/trial-lecture/model_experiment.svg)
]

.pull-right-42[
### Cancer Modeling popularity

![](img/trial-lecture/pubmed_cancer.png)

.purple.smaller[Brady &amp; Enderling (2019). Mathematical Models of Cancer: When to Predict Novel Therapies, and When Not to. Bull Math Biol 81, 3722–3731 (2019). https://doi.org/10.1007/s11538-019-00640-x]]

&lt;br/&gt;&amp;nbsp;&lt;br/&gt;

.little.purple[Rashevsky (1940). Mathematical biophysics in its relation to the cancer problem. Acta Biotheoretica, 5(3), 139–154. https://doi.org/10.1007/BF01567693]

???

- Model/experiment refinement cycle is a cornerstone of Systems Biology approaches 
- Models are a mix of **Mathematics** (theory) and **computational** (software) tools
- Model-experiment (data production) iterative refinement pattern/cycle
  - Use models can make new predictions and drive new hypotheses for further experimental investigations (**faster and with less cost**) and **guide us towards more informative experiments**
  - 2 **optimization** self-loops can be used in both models and experiments nodes
- Rashevsky (1940) talks about:
1. The usefulness of theoretical research for experimental science
2. Mathematical study of cell division and its relation to cancer

---

# Goals of computational cancer modeling

![:scale 85%](img/trial-lecture/goals_cancer_modeling.svg)

???

- Be **explanatory/mechanistic** and enable **translational medicine**
- Predict new therapies (e.g. **identify new targets** or synergistic drug combinations)
- Personalized treatment: **optimal drug-patient match**, optimize individualized cancer therapy
- Patient stratification (biomarkers) - **overcome patient diversity** (*Whe does a particular treatment work only in a fraction of patients?*), more accurate diagnostic tests
- Optimize clinical trial design (e.g. **better dosing schedules, optimize chemotherapy delivery, reduce drug toxicity**)
- Deliver robust and reproducible results
  - **VVUQ**: verification, validation (match observations - multiomics/patient data) and uncertainty quantification .small.purple[Groen (2021)]
  - **Patient-specific** results!
- **Data Integration and AI** =&gt; mention the promise of **IBM’s Watson**: applications to radiology and diagnoses aided by wearable technology amongst many others. Each of these examples are exciting, but **the full promise of personalized medicine remains unrealized**. While the amount and types of clinical data being generated for each cancer patient is increasing dramatically, there are no holistic approaches or algorithms available that can incorporate all this data to identify the best treatment for each individual patient.

---

# Modeling frameworks: an overview

.pull-left-58[
![](img/trial-lecture/multiscale.jpeg)
]

.pull-right-42[
### &amp;nbsp; &amp;nbsp; Aspects to consider

- System state and treatment of time

- Spatial information

- Stochasticity of reactions

- Black-box vs White-box

- Structure vs dynamical

- Area of application
]

&lt;br/&gt;&amp;nbsp;&lt;br/&gt;

- .green[Biological detail] <i class="fa fa-arrow-up" role="presentation" aria-label="arrow-up icon"></i> = .red[Data] + .red[Computation Demand] <i class="fa fa-arrow-up" role="presentation" aria-label="arrow-up icon"></i>
- Moving towards **multiscale hybrid modeling**

???

### PDEs and Stochastic models

.small[
- **Partial Differential Equations (PDEs)**
  - Time + space as unknown variables, .red[hard to solve in general]
  - Example 1: **Calcium (Ca2+)**, a fundamental intracellular signal, that achieves specificity over wide temporal and spatial changes
  - Example 2: Cell geometry, compartments (e.g. in whole-cell modeling)
- **Stochastic modeling**
  - Reaction processes are not deterministic .green[More realistic models]
  - .green[Molecular quantities are low] (#species &lt;100-1000)
  - .red[Need to solve the even harder **CME (Chemical Master Equation)**]: distribution of species across time and space
  - Approximate solutions: stochastic simulation algorithm (**Gillespie** algorithm + variations)
  - Particle-based methods (MD: **Molecular Dynamics**: solve equations for every atom in the system!)
]

### Structural/network vs dynamical/mechanistic models
  
.small[
- Structural models are network-based (connectivity)
  - .green[Exploit topological properties (e.g. graph theory) to derive associations/prove hypotheses/make predictions]
- Dynamical are mathematical models (so networks + equations)
]

### White-box vs Black-box models
.small[
- White-box: **mechanistic/hypothesis-driven**, use knowledge
  - ODEs, Boolean modeling, simple statistical models (linear regression, decision trees)
- Black-box: **empirical/data-driven**, .red[use A LOT of data]
  - Random forests, NNs, Deep Learning
- Grey-box: new Neural Network approaches
]

### Area of application

.small[
- **Tumor growth**, metabolism, vasculature, **microenvironment, immunity**, invasion, metastasis, **treatment and resistance**, etc.
- Important for considering the modeling framework
]

### Multiscale hybrid modeling

.small[
- Moving from **scale-specific** to **multiscale models**
- Incorporating different biological scales is important for **clinical translation**
- **Software** for multiscale modeling:
  - .green[Example]: Use ODEs or logical models for intracellular signaling and agent-based modeling (ABM) for interactions between cells (**PhysiBoSS** = PhysiCell + MaBoSS, see .purple[Letort (2018)])
  - **COPASI** (.purple[Hoops (2006)]): ODEs and stochastic simulations or hybrids of these
  - **Chaste: Cancer, Heart and Soft Tissue Environment** .purple[Cooper (2020)]
      - Cardiac electrophysiology
      - Cell-based modeling, application in tissue carcinogenesis
      - Multiscale lung ventilation simulations
  - **Virtual Cell** .purple[Blinov (2017)] (rule-based framework): combines ODEs + Stochastic + agent-based modeling.
]

---
class: center, middle

# .larger.blue[Mechanistic models]
----
## Why do we need them?

---

## Genomic data cannot explain complex cancer behavior

.pull-left-58[![:scale 100%,500px](img/trial-lecture/eduati_dataset.jpg)]
.pull-right-42[
&amp;nbsp; &amp;nbsp; .bold[Dataset] .small.purple[Eduati (2017)]
.little[
- 14 heterogeneous colon cancer cell lines
- 14 measured phosphoproteins
- 43 perturbed conditions
]
&amp;nbsp; &amp;nbsp; .bold[Observations]
.little[
- .red[No correlation] between perturbation response and CMS (*) or genotype data
- Signaling is .red[not determined by mutations]
- Use modeling to .green[gain insights] on signaling changes
]
&lt;p style='text-align: center;'&gt;.small[(*) .purple[Guinney (2015). The consensus molecular subtypes of colorectal cancer. Nature medicine, 21(11), 1350–1356. https://doi.org/10.1038/nm.3967]]&lt;/p&gt;
]

???

### A large-scale experimental perturbation dataset

- 14 **genetically heterogeneous** colorectal cancer cell lines grouped to 4 molecular subtypes (according to .purple[Guinney (2015)])
- Cell lines cover typical recurrent mutations in colorectal cancer (such as KRAS, TP53, APC, BRAF)
- 43 perturbed conditions (**7 inhibitors**, 5 ligand-stimulating receptors)
- 14 measured phoshoproteins
- Results differ strongly between cell lines
- **No clear correlation between genotype data or consensus subtype and perturbation response data**
- State and responsiveness of the signaling is not fully determined by mutations
- **Dataset provides information that is not contained in genomic data** 
  - Use of modeling to get insights on the network/signaling changes that lead to these heterogeneous responses

---

class: center, middle

# .larger.blue[Introduction to modeling formalisms]
----
## Boolean vs ODE modeling

---

![:scale 95%](img/trial-lecture/discrete_vs_continuous.svg)

???

Previous slide: we need modeling. Let's start with some basics first!

- .green[Attractors are **biologically meaningful**]:
  - Cell fates (phenotypes): normal cell state vs a diseased state
  - Stable patterns of cell activity (e.g. cell cycle or T cell differentiation)

---

# Summary

.pull-left[
### Logical Modeling .small.purple[Kauffman (1969)]

- Active/Inactive states (0/1)
- .red[Lack of representation of time]
- Less data needed for training and optimization
- .green[Attractors are biologically meaningful]
- Derivation of insights is challenging

&lt;br&gt;
.bold[Software]

.small[
.purple[Naldi (2018). Logical Modeling and Analysis of Cellular Regulatory Networks With GINsim 3.0]

.purple[Naldi (2018). The CoLoMoTo Interactive Notebook: Accessible and Reproducible Computational Analyses for Qualitative Biological Networks]
]]

.pull-right[
### ODE Modeling

- Species concentrations
- Time as continuous variable
- .red[Need for kinetic data] (rates, equilibrium constants)
  - Experiments
  - Parameter estimation
- More realistic dynamics
- .green[Model parameters directly link to mechanisms]

.bold[Software]

.small[
.purple[Hoops (2006). COPASI: a COmplex PAthway SImulator]

.purple[Choi (2018). Tellurium: An extensible python-based modeling environment for systems and synthetic biology]
]]

---

# Mixed Discrete-Continuous modeling

.pull-left-58[
### Logic-based ODE modeling .small.purple[Wittmann (2009)]

`\(\frac{dx_i}{dt} = \tau_i(B_i(f(x_{1,i}),...,f(x_{N,i})) - x_i)\)`

![:scale 60%](img/trial-lecture/eduati_parameters.jpg)]

.pull-right-42[

### .green[Assets]

.little[
- Minimum model complexity
- Interpretable parameters
- Capture signal strength `\((k)\)` and time dynamics `\((\tau)\)`
- No need for kinetic data
]

### Other mixed formalisms

.little[
- Fuzzy logic modeling .small.purple[Aldridge (2009)]
- Stochastic Boolean modeling (MaBoSS 2.0) .small.purple[Stoll (2017)]
]]

&lt;br/&gt;&amp;nbsp;&lt;br/&gt;

.small.purple[Gjerga (2020). Converting networks to predictive logic models from perturbation signalling data with CellNOpt]

???

- Each node is associated with a speed/responsiveness parameter `\(\tau\)`
- Each edge/interaction is associated with a strength parameter `\(k\)`

---

class: center, middle

# .larger.blue[Mechanistic models in practice]
----
## Building heterogeneous cancer models for personalized treatment

---

## .little[Cell-specificity achieved with logic-based ODE modeling]

.pull-left-58[
![](img/trial-lecture/eduati_model_heterogeneity.jpg)

.small.purple[Eduati (2017). Drug Resistance Mechanisms in Colorectal Cancer Dissected with Cell Type-Specific Dynamic Logic Models]]

.pull-right-42[
&amp;nbsp; &amp;nbsp; .red.bold[Drawbacks]

.little[
- Manually curated PKN
- Parameter Optimization
- Regularization
  - Prioritize good fit to data
  - Keep the model sparse
]

&amp;nbsp; &amp;nbsp; .green.bold[Assets]

.little[
- Cell line heterogeneity &lt;br&gt; 
`\((\)`different `\(k)\)`
- **One model per cell line**
]]

???

### Prior Knowledge Network (PKN)

- Manually curated
- `OmniPath` .purple[Türei (2016)] (49 nodes, 86 edges)
- Compress PKN to reduce model complexity without affecting logic consistency
  
### Heterogeneity of cell-specific models

- **More than half of the edge parameters were zero** =&gt; indicating signaling interactions not present in colon cancer (edge width denotes that)
- .green[Edge width in the PKN image shows the heterogeneity]: the associated numbers denote the proportion of pair-wise comparisons (out of `\(91={14 \choose 2}\)`) between cell lines, that have statistically different edge parameters `\((k\text{ 's})\)`
- Edge-parameter and tau-parameter heatmaps also show heterogeneity (Fig. 4c in the paper)

---

# .small[Model parameters elucidate drug sensitivity mechanisms]

.pull-left-58[
.bold[Parameter-drug associations (GDSC)]
![](img/trial-lecture/eduati_sensitive_parameters.jpg)

.bold[GSK3i with fixed MEKi]
![:scale 80%](img/trial-lecture/eduati_GSK3_inhibitors.png)

.little[.red.bold[Note:] Genomic data can't derive such associations!]
]
.pull-right-42[
![](img/trial-lecture/eduati_GSK3_param.jpg)
.green.bold[Assets]
- **Mechanism-based** therapeutic strategy
- **Personalized** combinatorial treatment
]

.purple[Eduati (2017)]

???

- Exploit drug response data from **GDSC** screening (drug sensitivity, IC50)
- Choose **27 drugs** (mostly ones that were tested in all 14 cell lines and target nodes in the PKN)
- Find if there are associations between cell line sensitivity for the tested drugs and the model parameters
- .green[Association 1: MEK-ERK parameters =&gt; EGFR signaling + BRAF signaling]
  - **Evidence** (in colorectal cancer): combination of EGFR (or BRAF) and MEK inhibitors has been suggested to overcome resistance to MEK inhibitors in BRAF and KRAS mutants
- .green[Association 2: GSK3 parameters =&gt; MEK inhibitors]
  - **Hypothesis**: colon cell lines resistant to MEK inhibitors and having high functional GSK3, will benefit from **GSK3 inhibition** (suggest new therapeutic strategies - they validated it experimentally)
- These associations couldn't be unveiled with genomic data!
- Such associations between model parameters and drug sensitivity are useful for personalized combinatorial therapies

---

## Cell-specificity in a large-scale ODE mechanistic model

.pull-left[
![:scale 85%](img/trial-lecture/frolich_model.jpg)

 .small.purple[Fröhlich (2018)]

- Curated, SBML format, .green[reusable]
- Model structure .red[uncertainty]
]

.pull-right[
![](img/trial-lecture/frolich_model_optim.jpg)

- .red[Parameter estimation] .smaller[(reaction rates)]
- .green[Scalable] optimization with parallel computing
- .bold.green[Cell-specificity] (120 cell lines)
  - Gene expression 
  - Somatic mutations
]

???

- Filling in the need for high-quality mechanistic large-scale ODE models
  - **Curated** from the meta-database `ConsensusPathDB` .purple[Herwig (2016)]
  - .green[Reusable] model, exported in green.[standard SBML format]
  - .green[Curated model]
  - .green[Model can be individualized to particular cell lines], so various cancer types and treatments
  - .red[Model structure uncertainty]: while the model is based on extensive literature and was manually curated, .red[interactions might still be missing or incorrect.]
  - .green[Suitable for prediction of drug sensitivity]
      - Includes the targets of 7 small molecular inhibitors
      - The authors suggest that .red[even larger models are needed] for a rigorous mechanistic description of the polypharmacology of anti-cancer drugs
- .red[Parameter estimation and optimization for cell-specificity (reaction rate constants)]: Parameterization of the model to **120 cell lines from CCLE** (using only gene expression and mutation data).
  - .green[Scalable and parallelized optimization] by an order of magnitude: data training with 400 cores **took &lt; 1 week** (compared to years that it would take with other methods)
  - The gradient evaluation time was dramatically reduced by using adjoint sensitivities, exploiting sparsity, and parallelization.

---

class: center, middle

# .larger.blue[Black-box vs White-box models]
----
## From performance to interpretability and the need for training data

---

### Mechanistic models vs standard Machine Learning methods &lt;br&gt; .small.purple[Fröhlich (2018)]

.pull-left-67[
.bold[Performance comparison]

![:scale 72%](img/trial-lecture/frolich_model_validation.jpg)

- .bold.green[Similar predictive power]
- No prior knowledge and parameterization required for ML methods
]

.pull-right-33[
.bold[Dataset split]

![:scale 70%](img/trial-lecture/frolich_train.jpg)

.little.bold[Dose-response predictions]

![:scale 70%](img/trial-lecture/frolich_predictions.jpg)
]

???

- **Dataset**: viability results of 7 drugs x 9 concentrations x 120 cell lines
- `d` defines up to which network distance interactions were considered in `lasso` method.
So this method took not only mutation and gene expression data, but also network topology as input!
- Mechanistic modeling performs equally well compared to other statistical methods (lasso, random forest for example).
**These methods top performers in previous drug sensitivity prediction DREAM challenges**
- Prediction of cell viability (single drug dose-response curves, DV90 is the cell line, PD{digits} is the drug)

---

## Molecular insights from mechanistic ODE modeling

.pull-left-67[
![](img/trial-lecture/frolich_insights1.jpg)
![](img/trial-lecture/frolich_insights2.svg)
]

.little.pull-right-33[

### &amp;nbsp; &amp;nbsp; Mechanistic models

- Parameters .green[link to cancer biology]
- Elucidate drug's MoA

### &amp;nbsp; &amp;nbsp; Statistical methods

- .red[No mechanistic insights]
- Do not exploit prior knowledge

&lt;br/&gt;&amp;nbsp;&lt;br/&gt;
.small.purple[Fröhlich (2018). Efficient Parameter Estimation Enables the Prediction of Drug Response Using a Mechanistic Pan-Cancer Pathway Model]
]

???

### Parameters link to cancer biology

- Parameters are reaction rates in these ODE models. Unlike parameters from statistical methods these provide **direct information about the underlying biological network**. Analysis showed large parameters uncertainties, which though didn't account for most of the errors in predicted viability (mostly due to prediction bias). Since studies have shown the fraction rates have lower uncertainty, they computed **equilibrium constants as the ratio of on (forward) and off (backward) rates of reversible reactions** (conversion of reactants to products and the conversion of products to reactants occurs simultaneously). .green[Parameters different from 1 are important for the observed drug sensitivity]. 9 such parameters were identified. Of these parameters, .green[5 out 9 link to cancer biology]:
  - 3 had to do with **receptor dimerization reactions** (or simply put complex formation), which are known to be important in cancer signal transduction
  - 1 enables the activation AKT, which is associated with increased cancer cell proliferation and survival in many cancer types
  - 1 is a binding linked to paradoxical activation, **driving the resistance in melanoma**

### Resistance and Susceptibility Markers Elucidate Drug Primary **Mode of Action (MoA)**

- Compute **cell-specific resistance (susceptibility) and sensitive biomarkers** using adjoint sensitivity analysis (sensitivity of output variables - cell viability - to input variables, positive/sensitive or negative/resistant)
- Derived a generalized set of candidate important markers (significant p-values) across all drugs used to build a simplified and high-level signaling network (a)
- Receptor targeting drugs revealed **similar mode of action (MoA)** and **similar classification of markers** within pathways [(b)-(d)]:
  - EGFR-SRC-STAT signaling responsible for **heterogeneity in drug sensitivity**
  - MAPK-PI3K signaling responsible for **heterogeneity in drug resistance**
  - .green[Information about signal flow] (.red[not possible with ML models])

---

.pull-left-42[
### Black-box models need training data!

![](img/trial-lecture/comboFM_dose_response.png)

Predicting response without drug combo data:

- ML methods .red[<i class="fa fa-times" role="presentation" aria-label="times icon"></i>] .small.purple[Julkunen (2020)]
- ODE models .green[<i class="fa fa-check" role="presentation" aria-label="check icon"></i>] .small.purple[Fröhlich (2018)]

.small.purple[Julkunen (2020). Leveraging multi-way interactions for systematic prediction of pre-clinical drug combination effects]
]

.pull-right-58[
### Accuracy vs Interpretability trade-off
![](img/trial-lecture/tradeoff.svg)
]

???

- To link to the VNN and DrugCell

### Predict drug combination response

- ML methods cannot predict drug combination data without being given drug combination data for training!
- **Latent means not experimentally measured**
- Another example of a very accurate drug combination prediction algorithm using tensor factorization method (order 5) that could outperform RFs is **comboFM** from .purple[Julkunen (2020)]. It predicts:
  - Missing entries in dose-response matrices
  - Dose-response of same two drugs, in a different cell line (tissue)
  - Dose-response of new drug combos in the same cell line
  - .red[But still they needed at least some drug-combo data to train on]!

---

## Interpretable Machine Learning: Grey-box modeling!

- Visual Neural Network model: DrugCell paper (white-box ML)

???

### New trend in AI research

Deep NNs have almost exhausted their utility (recognize objects, self-driving cars, beating world-champion at chess and Go), need for fresh approaches, going back to the roots (studying how babies learn, what is intelligence, how do we achieve generalization - take what you learn and use it in new situations, common sense, etc.). Towards this trend are efforts that make existing methods **more intepretable**, e.g. why Deep NNs make the decisions that they make.

This model, called DrugCell, was made mechanistically interpretable, or ‘visible’ by directly mapping the neurons of a deep neural network into a large hierarchy of known and putative molecular components and pathways of the human cell.

---

class: center, middle

# .larger.blue[Structural/Network models]
----
## Exploiting topological knowledge

---

## Topology-based measure identifies synergistic drug combinations for breast cancer

.pull-left-73[
<i class="fa fa-long-arrow-alt-right" role="presentation" aria-label="long-arrow-alt-right icon"></i> .bold.green[Hypothesis]: Cancer exploits cross-talk signaling to develop resistance

![](img/trial-lecture/jaeger_pci.jpg)

.small.purple[Jaeger (2017). Quantification of pathway cross-talk reveals novel synergistic drug combinations for breast cancer]
]

.pull-right-27[

### &amp;nbsp; &amp;nbsp; Analysis Steps

- Build network
- Compute PCI (\*)
- .little[PCI <i class="fa fa-arrow-up" role="presentation" aria-label="arrow-up icon"></i> = synergy <i class="fa fa-arrow-up" role="presentation" aria-label="arrow-up icon"></i>]

![](img/trial-lecture/jaeger_validation.jpg)

&amp;nbsp; &amp;nbsp; .small[(\*) PCI = **P**athway **C**ross- &lt;br&gt; &amp;nbsp; &amp;nbsp; &amp;nbsp; talk **I**nhibition]
]

???

- Pathway information from KEGG database
- Network efficiency (NE) is defined as the sum of **the inverse length of the shortest path between all network elements**. Topology 100%.
  - NE ranges between 0 and 1, where **1 indicates that all proteins communicate directly with each other**, that is, a fully connected network!
- PCI (pathway cross-talk inhibition) is the relative reduction in network efficiency
- **Right figure**: synergistic combinations (assessed in a previous screen experimentally) have a significantly higher PCI than non-synergistic drug pairs and particularly antagonistic ones
- .green[In silico analysis identified 390 novel anticancer drug pairs], likely to **diminish pathway cross-talk**
- .green[Authors experimentally validated selected drug combos, which showed high PCI in the analysis as synergistic]

---
class: center, middle

# .larger.blue[Modeling multi-cellular systems]
----
## Cellular Automata and Agent-based modeling

???

- **Cancer is never a single-cell disease!**
- Various types of cells in the tumor-microenvironment, each with a special role to play in the carcinogenesis.

---

## Cellular Automata: emergent complexity

![](img/trial-lecture/ca.svg)

.center.pull-left[
&lt;img src="https://media.giphy.com/media/d7SnByEMkrdeoVQ2lT/giphy.gif"&gt;&lt;/src&gt;

.purple[Game of Life, John Conway]
]

.pull-right[
.small.purple[Varela (2018). Epilog: A software for the logical modeling of epithelial dynamics]

  - Logical model framework for describing cell behavior

.purple.small[Cooper (2020). Chaste: Cancer, Heart and Soft Tissue Environment]

 - Model tissue carcinogenesis
]

---

## Two examples of Cellular Automata in cancer modeling

.pull-left[
.bold[Tumor-associated angiogenesis and vascular flow (2D)]

![:scale 80%](img/trial-lecture/ca_tumor_angiogenesis.jpeg)

.small.purple[McDougall (2006) Mathematical modelling of dynamic adaptive tumour-induced angiogenesis: Clinical implications and therapeutic targeting strategies]
]

.pull-right[
.bold[Modeling the interaction between TNBC and stromal cells (3D)]

![](img/trial-lecture/ca_stem_cells.jpeg)
.small.purple[Norton (2018) Modeling triple-negative breast cancer heterogeneity: Effects of stromal macrophages, fibroblasts and tumor vasculature]
]

???

1. (Left) =&gt; Tumors need nutrients and oxygen to proliferate. Angiogenesis is the process of generating new vasculature networks able to provide the required nutrients to tumors and thus enable their sustained growth. **Inducing angiogenesis is one of hallmarks of cancer**. The image shows the results from a mathematical model which simultaneously couples vessel growth with blood flow through the vessels—dynamic adaptive tumor-induced angiogenesis (DATIA). Authors studied this system in terms of how to best deliver chemotherapeutic drugs to the tumor. In this image, the parameters of the model are set as logically as possible (so-to-speak) and the .green[result is that the drug largely by-passes the tumor and recirculates to the parent vessel!]
2. (Right) =&gt; A 3D cellular automaton model (with stem-cell effects) of how chemical signaling with fibroblasts and macrophages can drive triple-negative breast cancer (TNBC). Triple-negative breast cancer (TNBC) accounts for about 10-15% of all breast cancers. The term triple-negative breast cancer refers to the fact that **the cancer cells don't have estrogen or progesterone receptors and also don't make too much of the protein called HER2**.
Stem cells proliferated and differentiated into progenitor cells, and cancer cells exchanged chemical signals with fibroblasts and infiltrating macrophages. Among their results, they found that .red[**increasing the stromal effect on cancer cell proliferation**] decreased overall tumor size, whereas .green[**increasing the stromal effect on cancer cell migration**] increased tumor size.

---

## .little[Agent-based modeling: tumor growth in 3D]

.pull-left[
.large.bold[Agent-Based Modeling]

.green.bold[Assets]

- Lattice-free
- Autonomous agents
- Incorporate space and stochasticity
- Utilize parallelism
- Reproduce emergent behavior
- Intuitive for multi-cellular modeling

.red.bold[Drawbacks]

- Lack of mechanistic interpretation
- Computationally intensive
]

.pull-right[
&lt;iframe width="100%" height="300" src="https://www.youtube.com/embed/16EyDBf0l_M" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen&gt;&lt;/iframe&gt;

.blue.bold[Spatiotemporal dynamics]

.little[
- Loss of tumor symmetry and homogeneity
- Cell mechanics drive proliferation
- .red[2 weeks] worth of calculations!
]]

.small.purple[Ghaffarizadeh (2018). PhysiCell: An open source physics-based cell simulator for 3-D multicellular systems]

???

### Agent-Based Modeling (ABM)

.small[
- **Lattice-free** modeling (compared to CA, which are mostly 2D, grid-based only).
- **Autonomous agents** freely interacting with one another over time and space by exchanging messages
- Environment is non-deterministic, so **agents have to learn and adapt their behavior** to achieve their goals
- .green[easily incorporate space, stochasticity, utilize parallelism]
- .green[reproduce *emergent properties*] (dynamics not possible to infer from agents alone)
- .green[ABM is intuitive for multi-cellular modeling (cells as agents) and to study problems such as tissue engineering evolution of bacterial colonies, **tumor metastasis**]
- .red[Complex nature of agent interactions =&gt; lack of mechanistic interpretation] (cannot apply formal analysis to explain system behavior from agents' rules)
  - Equation-based methods have several techniques to do that (parameter sensitivity, etc.)
- .red[Very computationally intensive] (tracking agents across space and time)
]

### Description of video and observations

- Cancer cells each have an individual expression of a .red[mutant oncoprotein], which drives proliferation. Yellow cells divide faster (lots of oncoprotein) than blue ones (very little oncoprotein). **All cancer cells consume oxygen!!!**
- .red[Oxygen transport limits] have lead to the formation of **a necrotic core (brown central region)**
- .green[Spatiotemporal dynamics]
  - Starting from a random "salt and pepper" distribution of cell proliferation rates to **more homogeneous** patches of yellow color, more or less showing **clonal expansion**
  - However, this effect is not uniform: in slower-growing regions with less growth substrate (oxygen) to drive proliferation, the "salt and pepper" distribution remains much longer, particularly near the necrotic core.
  - The mechanics drive an interesting additional angle. Mechanics-free models (e.g., cellular automata) typically predict that clones grow out like sectors in a circle (pizzas). However, if you add the mechanics, the faster-growing clones are more circular, because **they can push inward as well as outward**. Moreover, this means that once .green[faster-growing sub-populations reach the outer edge of the tumor with the most resources, they start growing to surround the slower populations and eventually push them towards the necrotic core and ultimately to extinction.] This seems to accelerate selection.

---

# Simulation of cancer immunotherapy

.pull-left-67[
![](img/trial-lecture/physicell_cancer_immunology.png)

.small.purple[Ghaffarizadeh (2018). PhysiCell: An open source physics-based cell simulator for 3-D multicellular systems]

<i class="fa fa-long-arrow-alt-right" role="presentation" aria-label="long-arrow-alt-right icon"></i> Parameter optimization in 2D simulations correlated with patient response to immunotherapy .smalli.purple[Ozik (2019). Learning-accelerated discovery of immune-tumour interactions]
]
.pull-right-33[
- A simple model of immune attack!
- Biased stochastic immune cell migration leads to insufficient cancer killing

&amp;nbsp; &amp;nbsp;.red.bold[Drawbacks]
- Many simplifications
- Need for more simulations
- Computational cost
]

???

## Exploring the spatial dynamics of stochastic tumor-immune contact interactions

### Video description

To model immunosurveillance, after simulating 14 days of growth we introduced generic immune cell agents that move towards tumor cells by **chemotaxis** (a random biased migration towards a tumor-cell-released chemical factor), test for contact with cells, stochastically form spring-like adhesions to any cell in close contact, and then test for **immunogenicity**. While adhered to a target cell, the immune cell agent attempts to induce apoptosis (e.g., by the FAS receptor pathway35) with a probability that scales linearly with immunogenicity (**higher oncoprotein, higher immunogenicity, higher chance to induce apoptosis**). If successful, the tumor cell undergoes apoptosis, while the immune agent detaches and resumes its chemotactic search for additional tumor cell targets. If the immune cell does not kill the tumor cell, it remains attached while making further attempts to induce apoptosis until either succeeding or reaching a maximum attachment lifetime, after which it detaches without inducing apoptosis.

- Because immune cell migration was strongly biased along the gradient of the immunogenicity chemical factor, they pass by some tumor cells at the outer periphery. .green[These surviving cells repopulate the tumor].

- .green[We found that stochastic immune cell migration had a major impact on this system] by increasing spatial mixing between tumor and immune cells, **potentially contributing to more successful immune responses.** Less biased migration would increase mixing of cancer and immune cells and increase the efficacy of the immune attack.

### Limitations

- Generic immune cells (not specific, like T cell or dendritic cell)
- .red[To further understand this system would require hundreds or thousands of additional simulations.].

## Follow-up study (parameter grid search in 2D)

In a follow-up study the authors used **HPC architecture workflows and Evolutionary Algorithms** to optimally search a design parameter space (6 parameters, example: .green[oncoprotein threshold, immune cell kill rate, migration rate, apoptosis rate], etc.) and discover optimal cancer regression regions within biological and clinical constraints, thus paving the way for **improvements in cancer immunotherapy (kill more cells in the 2D simulations)**.

### Limitations

- Their results explained the variation in patient response (2-20%).
.green[The larger the simulated fraction of cancer cells killed, the smaller the viable design space was!]
- .red[No identification of molecular mechanisms that lead to cell behaviors and no 3D]

---
class: center, middle

# .larger.blue[Summary]
----
### Things to consider when choosing a cancer modeling framework

Knowledge of assets and drawbacks .green[<i class="fa fa-smile" role="presentation" aria-label="smile icon"></i>]

Availability of Prior Knowledge (model size)

The type of information available (data)

The granularity of answers sought from the simulations

Area of application

Familiarity with the appropriate software

???

- Discuss e.g. benefits/ramifications for the clinic  - realistic data availability under those conditions, possible treatment suggestions for a clinician? Value for pre-clinical research (mechanisms and in silico screens) and clinical research
- Maybe assess each approach against a possible pre-clinical or clinical application, this will make very clear the assets and drawbacks?

---

## References (I)

### General

.littler[
- .purple[Kauffman, S. A. (1969)]. Metabolic stability and epigenesis in randomly constructed genetic nets. Journal of Theoretical Biology, 22(3), 437–467. https://doi.org/10.1016/0022-5193(69)90015-0
- .purple[Morris et al. (2010)]. Logic-based models for the analysis of cell signaling networks. Biochemistry, 49(15), 3216–3224. https://doi.org/10.1021/bi902202q
- .purple[Hanahan &amp; Weinberg (2011)]. Hallmarks of cancer: the next generation. Cell, 144(5), 646–674. https://doi.org/10.1016/j.cell.2011.02.013
- .purple[ElKalaawy et al. (2015)]. Methodologies for the modeling and simulation of biochemical networks, illustrated for signal transduction pathways: A primer. Biosystems, 129, 1–18. https://doi.org/10.1016/J.BIOSYSTEMS.2015.01.008
- .purple[Le Novere, N. (2015)]. Quantitative and logic modelling of molecular and gene networks. Nature Reviews Genetics, 16(3), 146–158. https://doi.org/10.1038/nrg3885
- .purple[Senft et al. (2017)]. Precision Oncology: The Road Ahead. Trends in Molecular Medicine, 23(10), 874–898. https://doi.org/10.1016/j.molmed.2017.08.003
- .purple[Apweiler et al. (2018)]. Whither systems medicine? Experimental &amp; Molecular Medicine, 50(3), e453. https://doi.org/10.1038/emm.2017.290
- .purple[Metzcar et al. (2019)]. A Review of Cell-Based Computational Modeling in Cancer Biology. JCO Clinical Cancer Informatics, (3), 1–13. https://doi.org/10.1200/CCI.18.00069
- .purple[Bekisz &amp; Geris (2020)]. Cancer modeling: From mechanistic to data-driven approaches, and from fundamental insights to clinical applications. Journal of Computational Science, 46, 101198. https://doi.org/10.1016/J.JOCS.2020.101198
]

???

Mostly **reviews** or papers that didn't fit to other categories

---

## References (II)

### Example Applications

.small[
- .purple[McDougall et al. (2006)]. Mathematical modelling of dynamic adaptive tumour-induced angiogenesis: clinical implications and therapeutic targeting strategies. Journal of theoretical biology, 241(3), 564–589. https://doi.org/10.1016/j.jtbi.2005.12.022
- .purple[Wittmann et al. (2009)]. Transforming Boolean models to continuous models: methodology and application to T-cell receptor signaling. BMC Systems Biology, 3(1), 98. https://doi.org/10.1186/1752-0509-3-98
- .purple[Jaeger et al. (2017)]. Quantification of pathway cross-talk reveals novel synergistic drug combinations for breast cancer. Cancer Research, 77(2), 459–469. https://doi.org/10.1158/0008-5472.CAN-16-0097
- .purple[Eduati et al. (2017)]. Drug Resistance Mechanisms in Colorectal Cancer Dissected with Cell Type-Specific Dynamic Logic Models. Cancer Research, 77(12), 3364–3375. https://doi.org/10.1158/0008-5472.CAN-17-0078
- .purple[Fröhlich et al. (2018)]. Efficient Parameter Estimation Enables the Prediction of Drug Response Using a Mechanistic Pan-Cancer Pathway Model. Cell Systems, 7(6), 567-579.e6. https://doi.org/10.1016/J.CELS.2018.10.013
- .purple[Norton et al. (2018)]. Modeling triple-negative breast cancer heterogeneity: Effects of stromal macrophages, fibroblasts and tumor vasculature. Journal of theoretical biology, 452, 56–68. https://doi.org/10.1016/j.jtbi.2018.05.003
- .purple[Ozik et al. (2019)]. Learning-accelerated discovery of immune-tumour interactions. Molecular Systems Design &amp; Engineering, 4(4), 747–760. https://doi.org/10.1039/C9ME00036D
- .purple[Kuenzi et al. (2020)]. Predicting Drug Response and Synergy Using a Deep Learning Model of Human Cancer Cells. Cancer Cell, 38(5), 672–684. https://doi.org/10.1016/J.CCELL.2020.09.014
- .purple[Julkunen et al. (2020)]. Leveraging multi-way interactions for systematic prediction of pre-clinical drug combination effects. Nasture Communications, 11(1), 6136. https://doi.org/10.1038/s41467-020-19950-z
]

---

## References (III)

### Software

.small[
- .purple[Hoops et al. (2006)]. COPASI--a COmplex PAthway SImulator. Bioinformatics, 22(24), 3067–3074. https://doi.org/10.1093/bioinformatics/btl485
- .purple[Aldridge et al. (2009)]. Fuzzy logic analysis of kinase pathway crosstalk in TNF/EGF/insulin-induced signaling. PLoS Computational Biology, 5(4), e1000340. https://doi.org/10.1371/journal.pcbi.1000340
- .purple[Blinov et al. (2017)]. Compartmental and Spatial Rule-Based Modeling with Virtual Cell. Biophysical Journal, 113(7), 1365–1372. https://doi.org/10.1016/j.bpj.2017.08.022
- .purple[Stoll et al. (2017)]. MaBoSS 2.0: an environment for stochastic Boolean modeling. Bioinformatics, 33(14), 2226–2228. https://doi.org/10.1093/bioinformatics/btx123
- .purple[Varela et al. (2018)]. Epilog: A software for the logical modelling of epithelial dynamics [version 2; peer review: 3 approved]. F1000Research, 7, 1145. https://doi.org/10.12688/F1000RESEARCH.15613.2
- .purple[Ghaffarizadeh et al. (2018)]. PhysiCell: An open source physics-based cell simulator for 3-D multicellular systems. PLoS Computational Biology, 14(2). https://doi.org/10.1371/journal.pcbi.1005991
- .purple[Letort et al. (2018)]. PhysiBoSS: a multi-scale agent-based modelling framework integrating physical dimension and cell signalling. Bioinformatics, 35(7), 1188–1196. https://doi.org/10.1093/bioinformatics/bty766
- .purple[Naldi et al. (2018)]. Logical Modeling and Analysis of Cellular Regulatory Networks With GINsim 3.0. Frontiers in Physiology, 9, 646. https://doi.org/10.3389/fphys.2018.00646
- .purple[Naldi et al. (2018)]. The CoLoMoTo Interactive Notebook: Accessible and Reproducible Computational Analyses for Qualitative Biological Networks. Frontiers in Physiology, 9, 680. https://doi.org/10.3389/fphys.2018.00680
- .purple[Choi et al. (2018)]. Tellurium: An extensible python-based modeling environment for systems and synthetic biology. BioSystems, 171, 74–79. https://doi.org/10.1016/j.biosystems.2018.07.006
- .purple[Gjerga et al. (2020)]. Converting networks to predictive logic models from perturbation signalling data with CellNOpt. Bioinformatics. https://doi.org/10.1093/bioinformatics/btaa561
- .purple[Cooper et al. (2020)]. Chaste: Cancer, Heart and Soft Tissue Environment. Journal of Open Source Software, 5(47), 1848. https://doi.org/10.21105/joss.01848
]

---

class:center,middle

# .blue.large[Thanks!]
----
![](img/ack/questions.jpeg)
    </textarea>
<style data-target="print-only">@media screen {.remark-slide-container{display:block;}.remark-slide-scaler{box-shadow:none;}}</style>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script src="macros.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function(d) {
  var s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})(document);

(function(d) {
  var el = d.getElementsByClassName("remark-slides-area");
  if (!el) return;
  var slide, slides = slideshow.getSlides(), els = el[0].children;
  for (var i = 1; i < slides.length; i++) {
    slide = slides[i];
    if (slide.properties.continued === "true" || slide.properties.count === "false") {
      els[i - 1].className += ' has-continuation';
    }
  }
  var s = d.createElement("style");
  s.type = "text/css"; s.innerHTML = "@media print { .has-continuation { display: none; } }";
  d.head.appendChild(s);
})(document);
// delete the temporary CSS (for displaying all slides initially) when the user
// starts to view slides
(function() {
  var deleted = false;
  slideshow.on('beforeShowSlide', function(slide) {
    if (deleted) return;
    var sheets = document.styleSheets, node;
    for (var i = 0; i < sheets.length; i++) {
      node = sheets[i].ownerNode;
      if (node.dataset["target"] !== "print-only") continue;
      node.parentNode.removeChild(node);
    }
    deleted = true;
  });
})();
(function() {
  "use strict"
  // Replace <script> tags in slides area to make them executable
  var scripts = document.querySelectorAll(
    '.remark-slides-area .remark-slide-container script'
  );
  if (!scripts.length) return;
  for (var i = 0; i < scripts.length; i++) {
    var s = document.createElement('script');
    var code = document.createTextNode(scripts[i].textContent);
    s.appendChild(code);
    var scriptAttrs = scripts[i].attributes;
    for (var j = 0; j < scriptAttrs.length; j++) {
      s.setAttribute(scriptAttrs[j].name, scriptAttrs[j].value);
    }
    scripts[i].parentElement.replaceChild(s, scripts[i]);
  }
})();
(function() {
  var links = document.getElementsByTagName('a');
  for (var i = 0; i < links.length; i++) {
    if (/^(https?:)?\/\//.test(links[i].getAttribute('href'))) {
      links[i].target = '_blank';
    }
  }
})();
// adds .remark-code-has-line-highlighted class to <pre> parent elements
// of code chunks containing highlighted lines with class .remark-code-line-highlighted
(function(d) {
  const hlines = d.querySelectorAll('.remark-code-line-highlighted');
  const preParents = [];
  const findPreParent = function(line, p = 0) {
    if (p > 1) return null; // traverse up no further than grandparent
    const el = line.parentElement;
    return el.tagName === "PRE" ? el : findPreParent(el, ++p);
  };

  for (let line of hlines) {
    let pre = findPreParent(line);
    if (pre && !preParents.includes(pre)) preParents.push(pre);
  }
  preParents.forEach(p => p.classList.add("remark-code-has-line-highlighted"));
})(document);</script>

<script>
slideshow._releaseMath = function(el) {
  var i, text, code, codes = el.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
};
slideshow._releaseMath(document);
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
